File: unit-3/sprint-3/day-2/we/app/src/main/java/com/example/intentservicedownloadfile/MainActivity.java
package com.example.intentservicedownloadfile;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

public class MainActivity extends AppCompatActivity {
    private Button mBtnDownloadFile;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Intent intent = new Intent(this, DownloadIntentService.class);
        mBtnDownloadFile = findViewById(R.id.btnDownload);
        mBtnDownloadFile.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startService(intent);
            }
        });
    }
}

File: unit-3/sprint-4/day-2/you/18AugPostRequest/app/src/main/java/com/example/a18augpostrequest_kotlin/MainActivity.kt
package com.example.a18augpostrequest_kotlin
import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_main.*
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class MainActivity : AppCompatActivity() {
    private lateinit var responseModel: ResponseModel

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        btnPost.setOnClickListener{
            progressbar.visibility = View.VISIBLE
            callApi()
        }
    }

    private fun callApi() {
        val apiService = Network.getInstance().create(ApiService::class.java)
        val requestType = etRequestType.text.toString()
        val email = etEmail.text.toString()
        val password = etPassword.text.toString()
        val title = etTitle.text.toString()
        val requestModel : RequestModel = RequestModel(email,password,requestType,title)
        apiService.postData(requestModel).enqueue(object : Callback<ResponseModel>{
            override fun onResponse(call: Call<ResponseModel>, response: Response<ResponseModel>) {

                response.body()?.run {
                    progressbar.visibility = View.GONE
                    Toast.makeText(applicationContext,"Data posted on Server",Toast.LENGTH_SHORT).show()
                    val intent = Intent(this@MainActivity,MainActivity2::class.java)
                    intent.putExtra("title",title)
                    intent.putExtra("email",email)
                    intent.putExtra("password",password)
                    intent.putExtra("requestType",requestType)
                    startActivity(intent)
                }
            }

            override fun onFailure(call: Call<ResponseModel>, t: Throwable) {
                Toast.makeText(applicationContext, "Data Posting Unsuccessful", Toast.LENGTH_SHORT).show()
            }

        })

    }


}